src/firebaseCredentials.js

```diff
@@ -0,0 +1,50 @@
+import firebase from "firebase/app";
+import "firebase/auth";
+import "firebase/firestore";
+
+export const firebaseConfig = {
+    apiKey: process.env.REACT_APP_FIREBASE_API_KEY,
+    authDomain: process.env.REACT_APP_FIREBASE_DOMAIN,
+    databaseURL: process.env.REACT_APP_FIREBASE_DATABASE_URL,
+    projectId: process.env.REACT_APP_FIREBASE_PROJECT_ID,
+    storageBucket: process.env.REACT_APP_FIREBASE_STORAGE_BUCKET,
+    messagingSenderId: process.env.REACT_APP_FIREBASE_SENDER_ID,
+    appId: process.env.REACT_APP_FIREBASE_APP_ID,
+    measurementId: process.env.REACT_APP_FIREBASE_MEASUREMENT_ID,
+};
+
+firebase.initializeApp(firebaseConfig);
+export const auth = firebase.auth();
+export const firestore = firebase.firestore();
+
+export const generateUserDocument = async (user, additionalData) => {
+    if (!user) return;
+    const userRef = firestore.doc(`users/${user.uid}`);
+    const snapshot = await userRef.get();
+    if (!snapshot.exists) {
+        const { email, displayName } = user;
+        try {
+            await userRef.set({
+                displayName,
+                email,
+                ...additionalData,
+            });
+        } catch (error) {
+            console.error("Error creating user document", error);
```

Comment:
Should we be logging errors to the console here and on line 48?

Reply:
oops, will remove this, I'm surprised the linters didn't pick this up

Reply:
Yup seems like we are allowing it based on the eslint file (https://github.com/uwblueprint/cdc-frontend/blob/master/.eslintrc#L37). Is that expected behaviour @ahmedhamodi?

Reply:
Yes console.errors are supposed to be allowed. Only log and info I think are caught by the linter, which is what I want. We can change that if the larger team disagrees though!

Reply:
Oh, info is also allowed based on the current eslint file. I'd personally vote to remove that, but honestly no one is using console.info unless they have a reason for it, in which case it is probably fine to allow. Logs are used much more often and are the ones that might leak sensitive information to a client.

Reply:
Cool, we can discuss that for later. For now, let's remove these errors since they are related to auth, just in case they contain anything sensitive :) @vivianliu0

Reply:
Sounds good, removed